sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -1578,9 +1593,15 @@ else if (!elemType.sameType(arg1.type.withMayBeNull(elemType.mayBeNull))) {
                 String method = getArrayOrMapCallName(call, arg0);
                 return new DBSPApplyExpression(node, method, type, arg0);
             }
+            case ARRAY_INSERT: {
+                validateArgCount(node, operationName, ops.size(), 3);
+                DBSPExpression op1 = ops.get(1)
+                        .cast(new DBSPTypeInteger(node, 32, true, ops.get(1).getType().mayBeNull));
+                String method = getArrayCallNameWithElemNullability(call, ops.get(0), op1, ops.get(2));
```

Comment:
This function generates the name of the rust function considering the nullability of the array elements.
We currently do not have a function `array_insert_N_N` (or the like, with the array elements being nullable). 

Maybe we should implement this after the Calcite bug has been fixed instead of adding a partial implementation? 
Or at least, fail to compile if the elements can be nullable (as it will likely produce a rust error anyway), and mention this in the docs.